### ISSUE TITLE
Increased pilot endpoint pushes after upgrade to GKE 1.23.8

### ISSUE THREAD

**Issue by bueti** [istio/istio#41914]
### Bug Description

We've upgraded a GKE Cluster from "1.22.12-gke.2300" to "1.23.8-gke.1900"

The moment the master nodes upgrade finished we saw a considerable increase of `pilot_push_triggers` of the endpoint type.
<img width="2482" alt="image" src="https://user-images.githubusercontent.com/383917/201025242-bcb23e2b-5869-400c-870f-872904a5c4bf.png">

24hr view:
<img width="2480" alt="image" src="https://user-images.githubusercontent.com/383917/201028391-b508d567-eaaf-4e08-813d-ac39fd60b7d1.png">

In addition we see an increase in `pilot_xds_config_size_bytes_sum`:
<img width="2487" alt="image" src="https://user-images.githubusercontent.com/383917/201025813-727af621-fcb6-406d-8763-9df143410c50.png">

The changes that are pushed are all incremental (so different issue than https://github.com/istio/istio/issues/39652). Simply the volume of changes increased dramatically.

Rate of "Push debounce"-logs:
<img width="2476" alt="image" src="https://user-images.githubusercontent.com/383917/201030001-962a141b-82dd-4948-8ced-94b4e7c4cd4a.png">

We do not see this behavior on an identical but much smaller cluster.

Any idea what is triggering this flood of endpoint changes?

### Version

```prose
$ istioctl version
client version: 1.15.3
control plane version: 1.14.3
data plane version: 1.14.3 (458 proxies)

$ kubectl version --short
Client Version: v1.23.6
Server Version: v1.23.8-gke.1900

$ helm version --short
v3.9.2+g1addefb
```

### Additional Information

I feel a bit uneasy to share the istioctl bug-report publicly. but i'm happy to share it privately or add any other debug info that could help pinpoint this problem.

**Comment by howardjohn** [comment]
Presumably endpoints started changing more regularly... maybe some pod is crash looping after the update. The logs of istiod show why its pushing, grep for `debounce`

**Comment by bueti** [comment]
> Presumably endpoints started changing more regularly... maybe some pod is crash looping after the update. The logs of istiod show why its pushing, grep for `debounce`

Thanks for taking the time to answer.
There are no crash looping pods or any other unhealthy behavior on the cluster that could explain the endpoint changes as far as I can tell.

looking for the debounce log messages I see a lot of similar changes in very short time:
<img width="1516" alt="image" src="https://user-images.githubusercontent.com/383917/201205360-d4d0c905-0a89-41da-ba98-90c743331849.png">

there are no corresponding changes on the pods that would explain these messages. From where does istio-discovery get the changes it propagates to the proxies?

**Comment by howardjohn** [comment]
I think that is just looking at a bunch of istiods. So its 1 event being handled by 10 pods. Id look a single istiod pod

**Comment by bueti** [comment]
makes sense, sorry. here the logs for one pod for 5 minutes:
[Explore-data-2022-11-10 22 08 55.csv](https://github.com/istio/istio/files/9984643/Explore-data-2022-11-10.22.08.55.csv)

**Comment by howardjohn** [comment]
Its still thanos-sidecar-grpc. I don't know what it is - you could try `kubectl get pod POD -w -oyaml` and see what is changing, or the same with the Service/Endpoints

**Comment by bueti** [comment]
Its late here, I’ll have a look tomorrow and see if I can pinpoint where these changes are coming from. 
its not just thanos, pretty much all deployments on the cluster are generating a lot more changes than before. The pods however are all in good shape and nothing changed except for the kubernetes version. 
on a cluster with the same workloads but smaller scale this does not happen.

**Comment by bueti** [comment]
> Its still thanos-sidecar-grpc. I don't know what it is - you could try `kubectl get pod POD -w -oyaml` and see what is changing, or the same with the Service/Endpoints

There don't seem to be any changes happening when istio discovers a change. Can you elaborate on how istio is notified about these changes? Is it listening to some kubernetes api?

**Comment by bueti** [comment]
I still couldn't figure out from where the increased config changes are coming. 

to visualize the increase of pilot pushes from k8s 1.22 vs 1.23. We get 10 times the changes that are pushed than before.

<img width="2145" alt="image" src="https://user-images.githubusercontent.com/383917/201367181-5f54a919-91d4-45c0-acd0-d0e4a7abb408.png">

**Comment by howardjohn** [comment]
It's watching the api server pod, endpoints, and service. It's almost certain one of those is changing.

**Comment by bueti** [comment]
I was able to narrow down the increase in changes to a configuration issue in one of our services[1]. Fixing the selector on the service reduced the volume of changes considerable.

I think this is related to [EndpointSlices](https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/) on Kubernetes. I noticed that the "classic" endpoint only returns targets that have the ip/port combo up, whereas endpoint slices have a list of all ips, no matter if the port is available or no. I'm still clarifying with Google Support if there are any meaningful changes between 1.22 and 1.23 on GKE. 

[1] the selector matched all nodes instead of a subset, thus we had over 100 matches but only a few of those actually had anything running the given port
